### ISSUE TITLE
TimerInfo has null ScheduleStatus if the interval is less than a minute

### ISSUE THREAD

**Issue by cg505** [Azure/azure-webjobs-sdk-extensions#436]
I was debugging my TimerTrigger function with a TimeSpan string of "00:00:10", and I was getting null reference exceptions when accessing TimerInfo.ScheduleStatus. Bumped it up to "00:01:00" and everything started working.

#### Repro steps

1. Add a function to your WebJob, like
```csharp
public void TestFunction([TimerTrigger("00:00:59", RunOnStartup = true, UseMonitor = true)] TimerInfo timer)
```

2. Run WebJob, notice that when `TestFunction()` is called, `timer.ScheduleStatus` is null.

3. Change TimeSpan string to `"00:01:00"`, notice issue is resolved.

#### Expected behavior

`timer.ScheduleStatus` is set.

#### Actual behavior

`timer.ScheduleStatus` is null.

#### Related information 

* Tried .NET 4.7 and 4.6.2
* WebJobs and WebJobs.Extensions 2.2.0
* I was able to reproduce the problem in a new project with no other code.

**Comment by mathewc** [comment]
Currently by design - for schedule frequencies of less than a minute, schedule monitoring is disabled (code [here](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/a3a5a49a64f3592c1eb7c1296086dc83c2011e96/src/WebJobs.Extensions/Extensions/Timers/Scheduling/TimerSchedule.cs#L76)).

I do see that you've explicitly set UserMonitor=true however, but we're overriding your setting based on our 1 minute cutoff logic.

We should either throw an error here, or not override the setting if user set. I think we should do the latter - do our existing defaulting if not set explicitly by the user, but if set don't override user setting.

**Comment by merfolk** [comment]
I came across this same problem. `ScheduleStatus` was null because I was using a 30 sec timer, but as soon as I raised it to 60 seconds the problem was resolved.

This 1 minute cutoff logic feels quite arbitrary and confusing. I would expect `ScheduleStatus` to be always populated.

What I really need though, is the time that my function last run _succesfully_, not just when it was last run regardless of outcome (success or failure).

**Comment by davidferro91** [comment]
Any update on this?

**Comment by dasiths** [comment]
I thought In was going crazy trying to figure out why it was null.

**Comment by MartinCa** [comment]
The documentation for timer functions explicitly mention that the default is only applied when `UseMonitor` is not explicitly set.
https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/functions-bindings-timer.md
> Set to true or false to indicate whether the schedule should be monitored. Schedule monitoring persists schedule occurrences to aid in ensuring the schedule is maintained correctly even when function app instances restart. If not set explicitly, the default is true for schedules that have a recurrence interval greater than or equal to 1 minute. For schedules that trigger more than once per minute, the default is false.

So either the code should be updated to match the documentation or the documentation updated to reflect that explicitly setting `UseMonitor` has no effect if the schedule interval is less than one minute.

**Comment by makokui** [comment]
Any update on this?
Is it likely that the code will be modified to work as described in the documentation, or will the documentation be modified along with the code?

**Comment by mathewc** [comment]
This just came up again in the alias. Moving back for Triage so we can make a change one way or another. At this point, I'm thinking we should just update the documentation to be explicit about this. Making a change to error or to no longer ignore would be a breaking behavioral change requiring a new major version.

**Comment by ccarmannt** [comment]
And, still not fixed more than a year later.

**Comment by mirekkukla** [comment]
Please either update the docs or the code. We burned a fair bit of time on this today.

**Comment by ericleigh007** [comment]
My suggestion, queue this up for a breaking change **in the code** when the major version comes out.  Until then, adjust the documentation.
I hate to see things that don't work, left broken in the name of compatibility.
The design we have now violates a golden rule of APIs "if a user tells you to do something, you don't change it silently"
If you're worried about compatbility, add a setting to the config so people can say whether that "override" bit if code gets executed or not if the user specifies.

**Comment by ericleigh007** [comment]
> The documentation for timer functions explicitly mention that the default is only applied when `UseMonitor` is not explicitly set. https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/functions-bindings-timer.md
> 
> > Set to true or false to indicate whether the schedule should be monitored. Schedule monitoring persists schedule occurrences to aid in ensuring the schedule is maintained correctly even when function app instances restart. If not set explicitly, the default is true for schedules that have a recurrence interval greater than or equal to 1 minute. For schedules that trigger more than once per minute, the default is false.
> 
> So either the code should be updated to match the documentation or the documentation updated to reflect that explicitly setting `UseMonitor` has no effect if the schedule interval is less than one minute.

Yep, it says the "default" is false.  I doesn't say "we overide what you tell us and always set false if the interval is less than a minute"
